app/src/main/java/com/sopt/now/presentation/signin/SignInViewModel.kt

```diff
@@ -0,0 +1,86 @@
+package com.sopt.now.presentation.signin
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.sopt.now.R
+import com.sopt.now.data.SharedPreferenceUtil
+import com.sopt.now.data.model.remote.request.RequestSignInDto
+import com.sopt.now.data.model.remote.response.ResponseSignInDto
+import com.sopt.now.data.ServicePool
+import dagger.hilt.android.lifecycle.HiltViewModel
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+import javax.inject.Inject
+
+@HiltViewModel
+class SignInViewModel @Inject constructor(private val sharedPreferences: SharedPreferenceUtil) :
+    ViewModel() {
+    private val _loginState: MutableLiveData<SignInState> = MutableLiveData(SignInState.Empty)
+    val loginState: LiveData<SignInState> get() = _loginState
+
+    private val id: MutableLiveData<String> = MutableLiveData("")
+    private val pw: MutableLiveData<String> = MutableLiveData("")
+
+    private val authService by lazy { ServicePool.authService }
+
+    init {
+        loginStateChange()
+    }
+
+    fun signIn() {
+        val signInRequest = getSignInRequestDto()
+        authService.signIn(signInRequest).enqueue(object : Callback<ResponseSignInDto> {
+            override fun onResponse(
+                call: Call<ResponseSignInDto>,
+                response: Response<ResponseSignInDto>,
+            ) {
+                if (response.isSuccessful) {
+                    val userId = response.headers()["location"]
+                    if (userId != null) {
+                        sharedPreferences.setUserId(USER_ID_KEY, userId.toInt())
+                    }
```

Comment:
저도 if문을 사용하긴 하지만 let문을 추천하는 분도 계시더라구요...!!
```suggestion
                    userId?.let { id ->
                        sharedPreferences.setUserId(USER_ID_KEY, id.toInt())
}
```

Reply:
솔직히 과제 급해서 노란줄 자동완성 눌러서 if문으로 되었습니다...ㅎㅎ